cooked-validators/src/Cooked/MockChain/Wallet.hs

```diff
@@ -101,59 +107,118 @@ txAddSignature w = Pl.addSignature' (walletSK w)
-    }
+initialTxFor initDist
+  | not $ validInitialDistribution initDist =
+    error "Not all UTxOs have at least minAda; this initial distribution is unusable"
+  | otherwise =
+    mempty
+      { Pl.txMint = mconcat (map (mconcat . snd) initDist'),
+        Pl.txOutputs = concatMap (\(w, vs) -> map (initUtxosFor w) vs) initDist'
+      }
   where
-    initDist' = M.toList $ distribution initDist
+    initUtxosFor w v = Pl.TxOut (walletAddress w) v Nothing
 
--- "Quick" currency is a convenience to manipulate assets that are supposed to
--- be already in the wild when running a mock chain. For example, a market
--- maker would exchange Ada against other assets called "coins". Defining a
--- minting policy for those coins is tedious and not interesting. The following
--- functions make it easy to manipulate such assets identified by a token name.
---
--- For example, `runMockChainWithDistribution (initialDistribution' [wallet 1,
--- quickValue "coin" 50])` provides 50 coins to wallet 1 alongside the default
--- 100_000_000 lovelace in the initial state.
+    initDist' = M.toList $ distribution initDist
 
--- | Extension of the default initial distribution with additional value in
--- some wallets.
-initialDistribution' :: [(Wallet, Pl.Value)] -> InitialDistribution
-initialDistribution' = (def <>) . distributionFromList
+-- * Quick Values
 
--- | The currency symbol of the "quick" is empty. Like Ada.
-quickCurrencySymbol :: Pl.CurrencySymbol
-quickCurrencySymbol = Pl.CurrencySymbol ""
+-- $quickvalues
+-- /Quick/ values are a convenience to manipulate assets that are supposed to
+-- be in existence when running a mock chain. For example, a market
+-- maker would exchange Ada against other assets. Yet, when writing traces
+-- for such contract we'd need to define a minting policy for those tokens,
+-- which is very repetivie, moreover, most of the times we'd want wallets to
+-- start with some positive valance of tokens.
+--
+-- There are two classes of functions for using custom tokens on traces:
+--
+-- 1. The @quick@ prefixed functions provide access to tokens from the
+--    @const (const True)@ minting policy. That is, these can be minted and
+--    burnt at will, at any point in time.
+--
+-- 2. The @perpetual@ prefixed functions provide access to tokens from
+--    the @const (const False)@ minting policy. That is, these /cannot/ ever
+--    be minted or burnt and must be present in an initial distribution to be useful.
+--
+-- See the docs for 'InitialDistribution' for an example usage.
 
--- | Token name of a "quick" asset class
+-- | Token name of a /quick/ asset class; prefixes the name with a @'q'@ to
+-- make it easy to distinguish between quick and permanent tokens.
 quickTokenName :: String -> Pl.TokenName
-quickTokenName = Pl.TokenName . Pl.stringToBuiltinByteString
+quickTokenName = Pl.TokenName . Pl.stringToBuiltinByteString . ("q" ++)
```

Comment:
I think it should be done when pretty printing in `UtxoState`.

For the first iteration of quick values which used the empty currency symbol, a special case was introduced in `UtxoState` to pretty print quick values (dropping the display of currency symbol). That piece of code is no longer useful because we now use the true/false minting policies which have non empty currency symbols. 

Instead of modifying the token names, I think we should match on the currency symbol in pretty printing in `UtxoState` and replace the currency symbol hash by "quick" or "permanent" in the display when it corresponds to the true or false minting policy.

Reply:
That makes sense, thanks for bringing this up. I'll implement